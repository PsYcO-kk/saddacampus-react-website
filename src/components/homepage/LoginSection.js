import React from 'react';
import NumberForm from './forms/NumberForm.js';
import OtpForm from './forms/OtpForm.js';
import ProfileSettings from './forms/ProfileSettings.js';
import "../../scripts/typewriter.js";

const typewriterStyle = {
	"color": "#e8c900"
};

const loaderStyle = {
	"color": "#555"
};

export default class LoginSection extends React.Component{
	constructor(){
		super();
		this.state = {
			NumberForm: true,
			OtpForm: false,
			ProfileSettings: false,
			number: '',
			country_code: '',
			otp: '',
			message: ''
		}
	}
	changeView = (newState) => {
		this.setState(() => newState);
	}
    render(){
        return(
			<section className="section-gap">
				<div className="container">
					<div className="row align-items-center">
						<div className="col-lg-6 about-left">
							<div className="jumbotron bg-white">
								<h1 className="display-4">Making college life <span className="wrap typewrite" data-period="2000" data-type='[ "easier...", "smarter...", "better..."]' style={typewriterStyle}></span></h1>
								<br/>
								<table id="loader">
									<tbody>
										<tr>
											<td><div id="sync"></div></td>
											<td>
												<h5 style={loaderStyle}>&nbsp; Loading <span className="wrap typewrite" data-period="500" data-type='[ "..."]'></span></h5>
											</td>
										</tr>
									</tbody>
								</table>
								<br />
								{this.state.NumberForm && <NumberForm changeView={this.changeView} />}
								{this.state.OtpForm && <OtpForm changeView={this.changeView} message={this.state.message} country_code={this.state.country_code} number={this.state.number}/>}
								{this.state.ProfileSettings && <ProfileSettings />}
							</div>
						</div>
						<div className="col-lg-6 about-right">
							<img className="img-fluid" src="./assets/images/image.jpg" alt=""/>
						</div>
					</div>
				</div>
			</section>
        );
    }
}