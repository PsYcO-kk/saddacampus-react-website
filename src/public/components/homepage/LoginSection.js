import React from 'react';
import NumberForm from './forms/NumberForm.js';
import OtpForm from './forms/OtpForm.js';
import ProfileSettings from './forms/ProfileSettings.js';
import ErrorBoundary from '../ErrorBoundary';
import Typewriter from './Typewriter';

const loaderStyle = {
	"color": "#555"
};

export default class LoginSection extends React.Component{
	constructor(){
		super();
		this.state = {
			loading: false,
			NumberForm: true,
			OtpForm: false,
			ProfileSettings: false,
			country_code: '',
			number: '',
			otp: '',
			message: ''
		}
	}
	changeState = (newState) => {
		this.setState(() => newState);
	}
    render(){
        return(
			<section className="section-gap">
				<div className="container">
					<div className="row align-items-center">
						<div className="col-lg-6 about-left">
							<div className="jumbotron bg-white">
								<h1 className="display-4">Making college life <Typewriter strings={["easier...", "smarter...", "better..."]} typeSpeed={130} backSpeed={50} color={'#e8c900'} /></h1>
								<br/>
								{this.state.loading &&
									<table>
										<tbody>
											<tr>
												<td><div id="sync"></div></td>
												<td>
													<h5 style={loaderStyle}>&nbsp; Loading...</h5>
												</td>
											</tr>
										</tbody>
									</table>
								}
								<br />
								<ErrorBoundary>
									{this.state.NumberForm && <NumberForm changeState={this.changeState} />}
									{this.state.OtpForm && <OtpForm changeState={this.changeState} message={this.state.message} country_code={this.state.country_code} number={this.state.number}/>}
									{this.state.ProfileSettings && <ProfileSettings country_code={this.state.country_code} number={this.state.number} />}
								</ErrorBoundary>
							</div>
						</div>
						<div className="col-lg-6 about-right">
							<img className="img-fluid" src="./assets/images/image.jpg" alt=""/>
						</div>
					</div>
				</div>
			</section>
        );
    }
}
