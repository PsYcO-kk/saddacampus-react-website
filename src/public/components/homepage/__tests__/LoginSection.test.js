import {shallow} from 'enzyme';
import React from 'react';
import LoginSection from '../LoginSection';

test("should render LoginSection correctly" , ()=>{
    const wrapper = shallow(<LoginSection />)
    expect(wrapper).toMatchSnapshot();
})

test("should change state/view of LoginSection with provided values" , ()=>{
    const wrapper = shallow(<LoginSection />);
    const newState = {
		loading: false,
        NumberForm: false,
        OtpForm: false,
        ProfileSettings: true,
        number: "",
        country_code : "",
        otp:"",
        message:""
	}
	const change = {
		NumberForm: false,
		ProfileSettings: true
	}
    wrapper.find('NumberForm').prop('changeState')(change);
    expect(wrapper.state()).toEqual(newState);
})
