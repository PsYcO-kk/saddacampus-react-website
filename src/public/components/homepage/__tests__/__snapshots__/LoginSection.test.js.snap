// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginSection correctly 1`] = `
<section
  className="section-gap"
>
  <div
    className="container"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-lg-6 about-left"
      >
        <div
          className="jumbotron bg-white"
        >
          <h1
            className="display-4"
          >
            Making college life 
            <Typewriter
              backSpeed={50}
              color="#e8c900"
              strings={
                Array [
                  "easier...",
                  "smarter...",
                  "better...",
                ]
              }
              typeSpeed={130}
            />
          </h1>
          <br />
          <br />
          <ErrorBoundary>
            <NumberForm
              changeState={[Function]}
            />
          </ErrorBoundary>
        </div>
      </div>
      <div
        className="col-lg-6 about-right"
      >
        <img
          alt=""
          className="img-fluid"
          src="./assets/images/image.jpg"
        />
      </div>
    </div>
  </div>
</section>
`;
